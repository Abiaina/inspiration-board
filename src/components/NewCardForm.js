import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: '',
      emoji: '',
    };
  }

  onFieldChange = (event) => {
    const fieldName = event.target.name;
    const fieldValue = event.target.value;
    const updateState = {};
    updateState[fieldName] = fieldValue;
    this.setState(updateState);
  }

  clearForm = () => {
    this.setState({
      text: '',
      emoji: ''
    });
  }
  onFormSubmit = (event) => {
    event.preventDefault();
    this.props.addCard(this.state)
    this.clearForm()
  }


  render() {
    return (
      <form onSubmit={this.onFormSubmit} className="new-card-form">
        <div>
          <label htmlFor="text">Text: </label>
          <input
           name="text"
           value={this.state.text}
           onChange={this.onFieldChange}
           type="text" />
        </div>
        <div>
          <label htmlFor="emoji">Emoji: </label>
          <input
          name="emoji"
          value={this.state.emoji}
          onChange={this.onFieldChange}
          type="text" />
        </div>

        <input
          className="button success"
          type="submit"
          value="Inspire"
        />
      </form>
    );
  }
}

NewCardForm.propTypes = {
}

export default NewCardForm;
