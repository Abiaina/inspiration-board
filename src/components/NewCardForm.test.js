import React from 'react';
import NewCardForm from './NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it matches an existing snapshot', () => {
    const wrapper = shallow( <NewCardForm addCardCallback={() => {} } />);
    expect(wrapper).toMatchSnapshot();
  });

  test('when a user enters text in the text field the field is updated', () => {
    // Arrange
    /// can use shallow b/c there is no subfields in NewCardForm
    const wrapper = shallow( <NewCardForm
      addCardCallback={() => {} } />);

    //find the inputfield
    let nameField = wrapper.find('input[name="name"]');

    // Act
    //simulates the event we are listening for
    nameField.simulate('change', {
      target: {
        text: 'testing out a positive note',
        emoji: '',
      },
    });
    //forces the on change event
    wrapper.update();

    //b/c re-rendered in the DOM have to look for nameField again
    nameField = wrapper.find('input[name="name"]');
    //Assert
    //.props here is only getting the properties on the the inputfield
    expect(nameField.getElement().props.value.toEqual('Bob'))
  });

  test('when the user types on a field the value changes', () => {
    const wrapper = shallow( <NewCardForm
      addCardCallback={() => {} } />);

    //find the inputfield
    let nameField = wrapper.find('input[name="name"]');

    // Act
    //simulates the event we are listening for
    nameField.simulate('change', {
      target: {
        name: 'name',
        value: 'Bob',
      },
    });
    //forces the on change event
    wrapper.update();

    //b/c re-rendered in the DOM have to look for nameField again
    nameField = wrapper.find('input[name="name"]');
    //Assert
    //.props here is only getting the properties on the the inputfield
    expect(nameField.getElement().props.value.toEqual('Bob'))

  })
});
