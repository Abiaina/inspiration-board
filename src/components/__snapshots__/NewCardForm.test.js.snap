// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot 1`] = `
<form
  className="new-card-form"
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="text"
    >
      Text: 
    </label>
    <input
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="emoji"
    >
      Emoji: 
    </label>
    <input
      name="emoji"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <input
    className="button success"
    type="submit"
    value="Inspire"
  />
</form>
`;
